opsgenie/resource_opsgenie_incident_template.go

```diff
@@ -0,0 +1,227 @@
+				Required: true,
+			},
+			"impacted_services": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"stakeholder_properties": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enable": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"message": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"description": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringLenBetween(1, 15000),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceOpsgenieIncidentTemplateCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := incident.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	createRequest := &incident.CreateIncidentTemplateRequest{
+		Name:                  d.Get("name").(string),
+		Message:               d.Get("message").(string),
+		Description:           d.Get("description").(string),
+		Tags:                  expandOpsgenieIncidentTemplateTags(d.Get("tags").(*schema.Set)),
+		Details:               expandOpsgenieIncidentTemplateDetails(d.Get("details").(map[string]interface{})),
+		Priority:              incident.Priority(d.Get("priority").(string)),
+		ImpactedServices:      expandOpsgenieIncidentTemplateImpactedServices(d.Get("impacted_services").(*schema.Set)),
+		StakeholderProperties: expandOpsgenieIncidentTemplateStakeholderProperties(d.Get("stakeholder_properties").([]interface{})),
+	}
+	result, err := client.CreateIncidentTemplate(context.Background(), createRequest)
+	if err != nil {
+		return err
+	}
+	d.SetId(result.IncidentTemplateId)
+	return nil
+}
+
+func resourceOpsgenieIncidentTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := incident.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+	if err != nil {
+		return err
+	}
+	if result != nil {
+		type element map[string]interface{}
+		var elementList []element
+		for _, value := range result.IncidentTemplates["incidentTemplates"] {
+			e1 := element{
+				"name":                  value.Name,
+				"id":                    value.IncidentTemplateId,
+				"message":               value.Message,
+				"tags":                  value.Tags,
+				"description":           value.Description,
+				"details":               value.Details,
+				"priority":              value.Priority,
+				"stakeholderProperties": value.StakeholderProperties,
+				"impactedServices":      value.ImpactedServices,
+			}
+			elementList = append(elementList, e1)
+		}
+		marshalJson, _ := json.Marshal(elementList)
+		d.Set("incidentTemplates", string(marshalJson))
```

Comment:
why json here we dont have any incidentTemplate varialbe but we are setting it . Basically im keeping unnecessary data ?

Reply:
Json here because the data I was getting from the `parse` function of `client.go` from `Opsgenie-go-sdk-v2` was of type incident.IncidentTemplate similar to payload/body type []uint8.

Reply:
okey but we dont have any field named in incidentTemplates

Reply:
we dont need to set anything we dont manage

Reply:
Got the point. Done.